src/server.rs

```diff
@@ -314,9 +314,9 @@ where
     /// This can be used for Unix Domain Sockets, or TLS, etc.
     ///
     /// Returns a `Future` that can be executed on any runtime.
-    pub fn serve_incoming<I>(self, incoming: I) -> impl Future<Output = ()> + 'static
```

Comment:
I *think* by removing this `+ 'static` bound on the *return type*, it might break existing usage. It might break this:

```rust
fn inc() -> impl Stream<Item = Result<TcpStream, io::Error>> + Send + 'static {
    unimplemented!()
}

let srv = warp::serve(routes).serve_incoming(inc());
tokio::spawn(srv);
```

Reply:
I do not think that this will break something, `'static` lifetime on returning type only limiting the lifetime, after change `'static` still will be allowed (with other lifetimes). Just tried this code, compiled and executed without problem:
```rust
#![deny(warnings)]

use futures::{stream, Stream};
use tokio::{io, net::TcpStream};
use warp::Filter;

fn inc() -> impl Stream<Item = Result<TcpStream, io::Error>> + Send + 'static {
    stream::pending()
}

#[tokio::main]
async fn main() {
    pretty_env_logger::init();

    let routes = warp::get().map(warp::reply);
    warp::serve(routes).serve_incoming(inc()).await;
}
```

Reply:
Awesome, thanks for checking!